content/docs/command-reference/exp/branch.md

```diff
@@ -15,29 +15,29 @@ positional arguments:
 
 ## Description
 
-Makes a
-[Git branch](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
-off the last commit (`HEAD`) based on the given `experiment`, using the `branch`
-name provided. This makes the given experiment
-[persistent](/doc/user-guide/experiment-management#persistent-experiments) in
-the repo.
-
-In most cases this is similar to using `dvc exp apply` (applies the `experiment`
-results to the workspace) followed by Git branching and committing, except that
-`dvc exp branch` **does not** switch into the created `branch`. Equivalent to
-this:
-
-```bash
-[master] $ git checkout -b <branch>
-[branch] $ dvc exp apply <experiment>
-[branch] $ git add . && git commit
-[branch] $ git checkout master
-```
-
-For [checkpoints](/doc/command-reference/exp/run#checkpoints), the `experiment`
-(custom Git branch with multiple commits) is merged into the new `branch`.
-
-To switch into the new branch, use `git checkout branch` and `dvc checkout`.
+Makes a named Git
+[`branch`](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
+containing the target `experiment` (making it
+[persistent](/doc/user-guide/experiment-management#persistent-experiments)). For
+[checkpoint experiments](/doc/command-reference/exp/run#checkpoints), the new
+branch will contain multiple commits (the checkpoints).
+
+The new `branch` will be based on the experiment's parent commit (`HEAD` at the
+time that the experiment was run). Note that unlike `git branch`, DVC **does
+not** switch into the new `branch`.
+
+`dvc exp branch` is useful for several scenarios, including:
+
+- To make one or more experiments persistent without applying them to the
+  workspace, so they can be
+  [stored and shared](https://dvc.org/doc/use-cases/sharing-data-and-model-files)
+  in a normal Git+DVC workflow;
+- In cases where `dvc exp apply` fails;
```

Comment:
This case doesn't make sense to me, `exp branch` is not a replacement for `exp apply`. `exp branch` is for creating a Git branch that contains all of the commits in an experiment run.

`exp apply` is for applying the (merged) contents of a single experiment commit into my current workspace.

Reply:
How would you "apply" a past non-checkpoint experiment (once `HEAD` has moved)?

Reply:
You can't. The only way you can use `dvc exp apply` in this scenario is with
```
git checkout <old-head-commit>
exp apply <exp>
```

If by "apply" here you really mean that you want to merge or rebase the changes from an experiment onto your new HEAD, you have to do it manually with the appropriate git commands. see also: https://github.com/iterative/dvc/issues/5782

Reply:
What about something like
```console
$ dvc exp branch exp-12341 exp-branch
$ git merge exp-branch
```
? (half DVC half Git)

Reply:
yes that will work, but it has the side effect of creating a full Git branch containing your experiment. If you have a ton of experiments you want to test against your new HEAD in this way, this is not ideal, since you probably don't actually want full Git branches for every one of those experiments.

Reply:
For now maybe
```suggestion
- In cases where it's not possible to `dvc exp apply` an experiment directly;
```

Reply:
Rephrased in 68892454. WDYT? If it feels to forced, obvious, or indirect we could just remove it indeed. But then there's only 2 items in this bullet list and they kind of relate so would need to rewrite as a single scenario/motivation paragraph.

Reply:
I still don't really think that bullet point belongs in the list.

Honestly it would probably be better to just combine the last two bullet points into a single statement/paragraph as you suggested.

It should just be something like

> `dvc exp branch` is useful if you would like to an experiment persistent as a Git branch, and for preserving checkpoints as independent commits within that newly created Git branch.

Reply:
IMO the first bullet can be put into the above paragraph and the last two bullets removed. I think the `apply` comparison is a bit too much of a stretch, and the independent checkpoint commits are mentioned already above. There's no reason the use cases need to be exhaustive. Giving the primary usage seems natural to me.

Reply:
OK I made it into a regular p with only the first item (rephrased slightly).